src/rgw/rgw_sal_motr.cc

```diff
@@ -1894,67 +1891,33 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
   // we should consider adding object entry to GC before deleting the metadata.
   // Delete from the cache first.
   source->store->get_obj_meta_cache()->remove(dpp, delete_key);
-  
-  if (ent.meta.size == 0) {
-    if (source->have_instance() && ent.is_delete_marker())
-    {
-      result.delete_marker = true;
-      result.version_id = ent.key.instance;
-      ldpp_dout(dpp, 20) << __func__ << ": Deleting delete marker - key " << delete_key << dendl;
-      // deleting delete-marker
-      rc = source->store->do_idx_op_by_name(bucket_index_iname,
-                                            M0_IC_DEL, delete_key, bl);
-      if (rc < 0) {
-        ldpp_dout(dpp, 0) << "Failed to delete delete-marker's entry from bucket index. " << dendl;
-        return rc;
-      }
-      // if deleted-marker version is the latest version, 
-      // then update is-latest flag to true for previous version.
-      if(ent.is_current())
-      {
-        ldpp_dout(dpp, 20)<<__func__<< " Updating previous version entries " << dendl;
-        bool set_is_latest=true;
-        rc = source->update_version_entries(dpp, set_is_latest);
-        if (rc < 0)
-          return rc;
-      }
-      // delete null reference key
-      if(ent.key.instance == "null")
-        del_null_ref_key = true;
-    }
+
+  // for delete-object without version-id parameter case,
+  // if delete-marker is latest entry in bucket index, 
+  // then return 0 without deleting any object.
+  if (!source->have_instance() && ent.is_delete_marker()){
+    rc = 0;
```

Comment:
I think, if I remember correctly, aws keeps on adding deleteMarker for every delete request. I am not sure how that is useful. Just sharing an observation.

Reply:
Yes in AWS it will create multiple delete-markers, but we kept the behavior inline with CEPH-RGW , there only one delete-marker is created for one obj.